tsdb/docs/refs.md

```diff
@@ -0,0 +1,82 @@
+# An overview of different Series and Chunk reference types
+
+## Used by callers of TSDB
+
+| Location           | Series access               | Chunk access                                                             |
+|--------------------|-----------------------------|--------------------------------------------------------------------------|
+| Global interfaces  | SeriesRef(in postings list) | chunks.ChunkRef (ChunkReader interface, Meta.Ref)                        |
+| Head               | HeadSeriesRef (autoinc)     | HeadChunkRef (5/3B) -> could be head chunk or mmapped                    |
+| blocks             | BlockSeriesRef (sorted 16B) | BlockChunkRef (4/4B split) -< this one is for both external and internal |
```

Comment:
> What does "both external and internal" mean here? Because all the refs in a way gets passed around externally and also used internally.

I don't remember.  I think we discussed internal vs external in the context of maybe the tsdb package. Either way I will just remove this.  once we have proper types a simple grep or "go to implementation" search in editors will reveal how a type is used.